### PR TITLE
Convert lib/srv/discovery to use slog

### PR DESCRIPTION
**Pull request by rosstimothy**

### CODE REVIEW

File: lib/srv/discovery/fetchers/db/azure_redis.go [hunk 1]
```diff
@@ -45,22 +47,26 @@ func (p *azureRedisPlugin) GetServerLocation(server *armredis.ResourceInfo) stri
...
 		return nil
 	}
 
 	if !p.isAvailable(server) {
-		log.Debugf("The current status of Azure Redis server %q is %q. Skipping.",
-			azure.StringVal(server.Name),
-			azure.StringVal(server.Properties.ProvisioningState))
+		logger.DebugContext(ctx, "Skipping unavailable Azure Redis server",
+			"server", azure.StringVal(server.Name),
+			"provisioning_state", azure.StringVal(server.Properties.ProvisioningState),
```
Comment (ravicious) [inline lib/srv/discovery/fetchers/db/azure_redis.go:59]:
```suggestion
			"status", azure.StringVal(server.Properties.ProvisioningState),
```

(to be consistent with other places in this file and package.

File: lib/srv/discovery/fetchers/db/azure_redis_enterprise.go [hunk 2]
```diff
@@ -44,32 +46,32 @@ func (p *azureRedisEnterprisePlugin) GetServerLocation(server *azure.RedisEnterp
...
 		return nil
 	}
 
 	if !p.isAvailable(server) {
-		log.Debugf("The current status of Azure Redis Enterprise %v is %q. Skipping.",
-			server,
-			azure.StringVal(server.Properties.ProvisioningState),
+		logger.DebugContext(ctx, "Skipping unavailable Azure Redis Enterprise server",
+			"server", server,
+			"provisioning_state", azure.StringVal(server.Properties.ProvisioningState),
```
Comment (ravicious) [inline lib/srv/discovery/fetchers/db/azure_redis_enterprise.go:65]:
```suggestion
			"status", azure.StringVal(server.Properties.ProvisioningState),
```

File: lib/srv/discovery/fetchers/db/azure_redis_enterprise.go [hunk 3]
```diff
@@ -90,9 +92,9 @@ func (p *azureRedisEnterprisePlugin) isAvailable(server *azure.RedisEnterpriseDa
 		armredisenterprise.ProvisioningStateFailed:
 		return false
 	default:
-		logrus.Warnf("Unknown status type: %q. Assuming Azure Enterprise Redis %v is available.",
-			azure.StringVal(server.Properties.ProvisioningState),
-			server,
+		slog.WarnContext(context.Background(), "Assuming Azure Enterprise Redis with unknown status type is available",
+			"status_type", azure.StringVal(server.Properties.ProvisioningState),
```
Comment (ravicious) [inline lib/srv/discovery/fetchers/db/azure_redis_enterprise.go:96]:
```suggestion
			"status", azure.StringVal(server.Properties.ProvisioningState),
```

File: lib/srv/discovery/fetchers/eks.go [hunk 4]
```diff
@@ -360,7 +364,7 @@ func (a *eksFetcher) getMatchingKubeCluster(ctx context.Context, clusterName str
 
 	switch st := aws.StringValue(rsp.Cluster.Status); st {
 	case eks.ClusterStatusUpdating, eks.ClusterStatusActive:
-		a.Log.WithField("cluster_name", clusterName).Debugf("EKS cluster status is valid: %s", st)
+		a.Logger.DebugContext(ctx, "EKS cluster status is valid", "status", st, "cluster_name", clusterName)
```
Comment (ravicious) [inline lib/srv/discovery/fetchers/eks.go:367]:
```suggestion
		a.Logger.DebugContext(ctx, "EKS cluster status is valid", "status", st, "cluster", clusterName)
```

To match other calls in this file.

File: lib/srv/discovery/fetchers/kube_services.go [hunk 5]
```diff
@@ -131,7 +131,7 @@ func (f *KubeAppFetcher) getServices(ctx context.Context) ([]v1.Service, error)
 			} else if match {
 				result = append(result, s)
 			} else {
-				f.Log.WithField("service_name", s.Name).Debug("Service doesn't match labels.")
+				f.Logger.DebugContext(ctx, "Service doesn't match labels", "service_name", s.Name)
```
Comment (ravicious) [inline lib/srv/discovery/fetchers/kube_services.go:134]:
```suggestion
				f.Logger.DebugContext(ctx, "Service doesn't match labels", "service", s.Name)
```

To match other calls in this file.

**Review by ravicious — APPROVED** [review]

File: lib/srv/discovery/fetchers/db/aws_rds.go [hunk 6]
```diff
@@ -144,37 +148,43 @@ func (f *rdsAuroraClustersPlugin) GetDatabases(ctx context.Context, cfg *awsFetc
...
 	}
 	databases := make(types.Databases, 0, len(clusters))
 	for _, cluster := range clusters {
 		if !libcloudaws.IsRDSClusterSupported(cluster) {
-			cfg.Log.Debugf("Aurora cluster %q (engine mode %v, engine version %v) doesn't support IAM authentication. Skipping.",
-				aws.StringValue(cluster.DBClusterIdentifier),
-				aws.StringValue(cluster.EngineMode),
-				aws.StringValue(cluster.EngineVersion))
+			cfg.Logger.DebugContext(ctx, "Skipping Aurora cluster that does not support IAM authentication",
+				"instance", aws.StringValue(cluster.DBClusterIdentifier),
```
Comment (greedy52) [inline lib/srv/discovery/fetchers/db/aws_rds.go:159]:
```suggestion
				"cluster", aws.StringValue(cluster.DBClusterIdentifier),
```

File: lib/srv/discovery/fetchers/db/aws_rds.go [hunk 7]
```diff
@@ -144,37 +148,43 @@ func (f *rdsAuroraClustersPlugin) GetDatabases(ctx context.Context, cfg *awsFetc
...
+				"error", err,
+			)
 		}
 
 		dbs, err := common.NewDatabasesFromRDSCluster(cluster, rdsDBInstances)
 		if err != nil {
-			cfg.Log.Warnf("Could not convert RDS cluster %q to database resources: %v.",
-				aws.StringValue(cluster.DBClusterIdentifier), err)
+			cfg.Logger.WarnContext(ctx, "Could not convert RDS cluster to database resources",
```
Comment (greedy52) [inline lib/srv/discovery/fetchers/db/aws_rds.go:184]: same as above `cluster` instead of instance

**Review by greedy52 — APPROVED** [review]

**Pull request merged**